python/pyarrow/tests/test_csv.py

```diff
@@ -1972,6 +1972,39 @@ def test_write_csv_decimal(tmpdir, type_factory):
     assert out.column('col').cast(type) == table.column('col')
 
 
+@pytest.mark.parametrize("data_size", (
+    int(1E2),
+    int(1E4),
+    int(1E6)
```

Comment:
Why the parameterization? What is the intent here?

Reply:
From playing with it, `int(1E6)` is the one that actually reproduces the bug (results in a segfault on main). 

Is there value in also having `int(1E2)` or `int(1E4)` to make sure there wasn't a regression with relatively smaller binary arrays? If there isn't, we can remove those parameters.

Reply:
I don't think there is value in that, no.